apache/superset
# feat: Frontend tagging

### SUMMARY

This PR is my POC of frontend tagging for dashboards, charts, and queries.

Note: This was for a hackathon and is currently incomplete:
- Editing tags is broken, you can double click the tag to edit but it will break when you try to press enter to submit the change.
- The query tags do not persist yet
- The list view searches do not work yet
- The welcome page has a dropdown for tags but it is not to design spec
- Excuse the mess, I did not clean up my code before pushing this up so there may be some arbitrary commented code and console.logs laying around from trying to debug editing tags.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

#### Global search from list view (this exists in the Charts, Dashboards, and Saved Queries list views):
![Screen Shot 2022-07-07 at 11 56 24 AM](https://user-images.githubusercontent.com/55605634/177829030-9186e61f-5ac5-4b6a-97b2-397bd3fb9a69.png)

#### Tags from within a chart (this exists on charts, dashboards, and the results section of queries)
![Screen Shot 2022-07-07 at 11 58 54 AM](https://user-images.githubusercontent.com/55605634/177829301-2b74c404-61db-437f-bffb-03a21533d780.png)

### TESTING INSTRUCTIONS

- Go to a chart, dashboard, or make a query and look at the results
- Click `+ New Tag` and type a tag name
- Press enter and observe the new tag
- Go to the Welcome page and search for your tag under the Tag dropdown
- Observe your tagged item, you can click this and it will bring you to the item you tagged
- Press the `X` on the tag to delete it
- Search for your tag on the welcome page and observe that it no longer shows up since it is no longer tagged

### ADDITIONAL INFORMATION
